fix all rubocop error in frontend

holylance opened the pull request:
Yesterday, My fork master branch had a problem.
So, I rebase and merge with spree/master branch.
I think It is fine now.

And then, I run rubocop in frontend directory and fix all rubocop errors in frontend.

WitekL committed:
Created promotion seed

holylance committed:
fix all rubocop error in frontend (#5)

joshRpowell committed:
Remove :default from edit view

damianlegawiec committed:
Circle CI 2.0 support

kushniryb committed:
Fix RSpec's deprecation in Reimbursement model spec

kushniryb committed:
Add missing locales

damianlegawiec committed:
Merge pull request #8730 from spark-solutions/enhancement/rspec-deprecation

Fix RSpec's deprecation in Reimbursement model spec

damianlegawiec committed:
Merge pull request #8729 from spark-solutions/enhancement/store-locales

Add missing locales

bbonislawski commented on frontend/app/controllers/spree/checkout_controller.rb:
```diff
@@ -159,9 +157,7 @@ def before_payment
-      if try_spree_current_user && try_spree_current_user.respond_to?(:payment_sources)
-        @payment_sources = try_spree_current_user.payment_sources
-      end
+      @payment_sources = try_spree_current_user.payment_sources if try_spree_current_user && try_spree_current_user.respond_to?(:payment_sources)
```
you don't need `try_spree_current_user &&` here. `try_spree_current_user.respond_to?(:payment_sources)` will do the job since `nil` have method `respond_to?` so you can call it like this

holylance replied:
Okay I changed it.

bbonislawski commented on frontend/app/controllers/spree/checkout_controller.rb:
```diff
@@ -174,9 +170,7 @@ def add_store_credit_payments
-        if @order.payments.valid.sum(:amount) < @order.total
-          redirect_to checkout_state_path(@order.state) and return
-        end
+        (redirect_to checkout_state_path(@order.state) and return) if @order.payments.valid.sum(:amount) < @order.total
```
This looks terrible. Ignore rubocop and use previous version

holylance replied:
Ah, I understand. Okay I will undo.

bbonislawski commented on frontend/app/controllers/spree/orders_controller.rb:
```diff
@@ -92,7 +97,7 @@ def accurate_title
 
     def check_authorization
       order = Spree::Order.find_by(number: params[:id]) if params[:id].present?
```
`order = Spree::Order.find_by(number: params[:id]) || current_order` should do the job. drop this `if` statement and next line

holylance replied:
Okay I changed it.

bbonislawski commented on frontend/lib/spree/frontend/middleware/seo_assist.rb:
```diff
@@ -14,7 +14,8 @@ def call(env)
           # redirect requests using taxon id's to their permalinks
-          if !taxon_id.blank? && !taxon_id.is_a?(Hash) && taxon = Taxon.find(taxon_id)
+          taxon = Taxon.find(taxon_id)
+          if !taxon_id.blank? && !taxon_id.is_a?(Hash) && taxon
```
This change introduces change in logic, since `find` will fail if taxon_id is blank and next line won't be called at all

holylance replied:
Ah, I understand. Okay I will undo.

bbonislawski reviewed (changes requested):
Great job but there are few small issues.

holylance committed:
Refactoring front end (#6)

* fix all rubocop error in frontend

* fix some code with review

damianlegawiec committed:
Merge pull request #8731 from spark-solutions/circle-ci-2-0

Circle CI 2.0 support

dylanfisher committed:
Update jQuery element exist checks to use length instead of is('*')

This is the jQuery recommended way of testing whether an element exists:
https://learn.jquery.com/using-jquery-core/faq/how-do-i-test-whether-an-element-exists/

It's also about 55% faster to do it this way:
https://jsperf.com/jquery-is-vs-length-2

damianlegawiec committed:
Merge pull request #8733 from dylanfisher/master

Update jQuery element exist checks to use `length` instead of `is('*')`

bbonislawski committed:
Merge pull request #8726 from spark-solutions/create-promotion-seed

Created promotion seed

damianlegawiec committed:
Merge pull request #8715 from joshRpowell/fix/add-default-attribute

Store: add default to permitted attributes

holylance committed:
Merge branch 'master' of github.com:holylance/spree

holylance committed:
fix all rubocop error in frontend (#7)
